lib/hsa/hsaRfiUpdateBadInputs.cpp

```diff
@@ -51,12 +55,34 @@ hsaRfiUpdateBadInputs::~hsaRfiUpdateBadInputs() {
 int hsaRfiUpdateBadInputs::wait_on_precondition(int gpu_frame_id) {
     (void)gpu_frame_id;
 
-    uint8_t* frame =
-        wait_for_full_frame(_network_buf, unique_name.c_str(), _network_buf_precondition_id);
-    if (frame == nullptr)
-        return -1;
-
-    _network_buf_precondition_id = (_network_buf_precondition_id + 1) % _network_buf->num_frames;
+    // Check for bad input updates
+    if (first_pass) {
+        DEBUG("hsaRfiUpdateBadInputs::wait_on_precondition: Waiting for full frame...");
+        uint8_t* frame = wait_for_full_frame(_in_buf, unique_name.c_str(), _in_buf_precondition_id);
+        DEBUG("hsaRfiUpdateBadInputs::wait_on_precondition: Bad inputs update.");
+        if (frame == NULL)
+            return -1;
+        _in_buf_precondition_id = (_in_buf_precondition_id + 1) % _in_buf->num_frames;
+        first_pass = false;
+        frames_to_update = device.get_gpu_buffer_depth();
+        filling_frame = true;
+    } else {
+        // Check for new bad inputs only if filled all gpu frames (not currently filling frame)
+        if (!filling_frame) {
```

Comment:
`filling_frame` will never get set back to false.  You should use the check of `frame_copy_active` and `frames_to_update` from my branch.